.github/workflows/deploy_dbt_docs.yaml

```diff
@@ -0,0 +1,71 @@
+name: deploy-dbt-docs
+
+on:
+  workflow_run:
+    workflows: [build-and-test-dbt]
+    branches: [master, data-catalog]
+    types: [completed]
+
+jobs:
+  deploy-dbt-docs:
+    runs-on: ubuntu-latest
+    # These permissions are required to make a GitHub Pages deployment
+    permissions:
+      pages: write      # To deploy to Pages
+      id-token: write   # To verify the deployment comes from an valid source
+
+    steps:
+      # There is currently no way to early-exit a run with a success status
+      # in GitHub actions, so we have to mark each of these steps with the
+      # same `if` condition to restrict deploys to successful builds :(
+      # https://github.com/actions/runner/issues/662
+      - if: github.event.workflow_run.conclusion == 'success'
+        name: Checkout
+        uses: actions/checkout@v3
+
+      - if: github.event.workflow_run.conclusion == 'success'
+        name: Install dbt requirements
+        uses: ./.github/actions/install_dbt_requirements
+
+      - if: github.event.workflow_run.conclusion == 'success'
+        name: Load environment variables
+        uses: ./.github/actions/load_environment_variables
+
+      - if: github.event.workflow_run.conclusion == 'success'
+        name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_TO_ASSUME_ARN }}
+          aws-region: us-east-1
+
+      - if: github.event.workflow_run.conclusion == 'success'
+        name: Generate docs
+        # Target is currently set to CI because we expect this action to be
+        # run against the long-lived data-catalog branch, but we should change
+        # this to prod when we merge that branch into master
+        run: dbt docs generate --target ci
+        working-directory: ${{ env.PROJECT_DIR }}
+        shell: bash
+        env:
+          HEAD_REF: data-catalog
+
+      - if: github.event.workflow_run.conclusion == 'success'
+        name: Package doc files for upload
+        run: |
+          mkdir _site
+          for file in index.html catalog.json manifest.json; do
+            cp "target/$file" "_site/$file"
+          done
```

Comment:
The `target` directory includes a bunch of metadata about the run, stored in files that could be accessed via GitHub Pages if someone knew what to look for. I scanned these files and couldn't find anything that looked obviously sensitive, but my confidence in that threat assessment is low (the files can be big), so I'm playing it safe and only explicitly packaging up files that are necessary to serve the docs site.